### ISSUE TITLE
Internal error on commit when user.name and user.email not defined

### ISSUE THREAD

**Issue by BTBurke** [sdg-mit/gitless#126]
An internal error is thrown when you commit to a repo without first setting the git user and email:

```
✘ Some internal error occurred
  ➜ If you want to help, see http://gitless.com for info on how to report bugs and include the following information:
  
  0.8.5
  
  Traceback (most recent call last):
    File "gitless/cli/gl.py", line 82, in main
    File "gitless/cli/gl_commit.py", line 57, in main
    File "pygit2/config.py", line 129, in __getitem__
    File "pygit2/config.py", line 113, in _get_entry
  KeyError: u'user.name'
```

It's pretty easy to figure out what the problem is from the KeyError, but a more helpful error message might be nice.

**Comment by jack17529** [comment]
Is the bug in this code - https://github.com/sdg-mit/gitless/blob/master/gitless/core.py

**Comment by spderosso** [comment]
@jack17529 The bug is in [gl_commit.py](https://github.com/sdg-mit/gitless/blob/f09afe9afaa6b2f8d6d884be50b9afb1e8900340/gitless/cli/gl_commit.py#L57-L64) (in addition to checking that the variable has a valid value we also have to check that the value is set)

**Comment by spderosso** [comment]
fixed in 5bc6bf6717bc731988d01fbd47bb4d8699fc5089
